### PR TITLE
fix(eventstore): sub queries

### PR DESCRIPTION
**Pull request by adlerhurst**
It's now possible to add multiple queries to an eventstore-query. This should improve query performance if multiple aggregates are requested in one query.

### CODE REVIEW

File: internal/command/iam_idp_config_model.go [hunk 1]
```diff
@@ -26,18 +27,21 @@ func NewIAMIDPConfigWriteModel(configID string) *IAMIDPConfigWriteModel {
...
 
 func (wm *IAMIDPConfigWriteModel) Query() *eventstore.SearchQueryBuilder {
-	return eventstore.NewSearchQueryBuilder(eventstore.ColumnsEvent, iam.AggregateType).
-		AggregateIDs(wm.AggregateID).
+	return eventstore.NewSearchQueryBuilder(eventstore.ColumnsEvent).
 		ResourceOwner(wm.ResourceOwner).
+		AddQuery().
+		AggregateTypes(iam.AggregateType).
+		AggregateIDs(wm.AggregateID).
+		EventTypes(iam.FeaturesSetEventType).
```
Comment (livio-a) [inline internal/command/iam_idp_config_model.go:35]: please remove

File: internal/eventstore/repository/sql/query.go [hunk 2]
```diff
@@ -125,31 +123,36 @@ func eventsScanner(scanner scan, dest interface{}) (err error) {
...
+		subClauses := make([]string, len(filter))
+		for subIdx, f := range filter {
+			value := f.Value
+			switch value.(type) {
+			case []bool, []float64, []int64, []string, []repository.AggregateType, []repository.EventType, *[]bool, *[]float64, *[]int64, *[]string, *[]repository.AggregateType, *[]repository.EventType:
+				value = pq.Array(value)
+			case map[string]interface{}:
+				var err error
+				value, err = json.Marshal(value)
+				logging.Log("SQL-BSsNy").OnError(err).Warn("unable to marshal search value")
```
Comment (livio-a) [inline internal/eventstore/repository/sql/query.go:144]: i know this part is unchanged, but shouldn't we ignore the filter on error?

**Review by livio-a — CHANGES_REQUESTED** [review]

File: internal/eventstore/repository/sql/query.go (hunk 2 above)
Comment (adlerhurst) [reply internal/eventstore/repository/sql/query.go:144]: the clause will be ignored now

**Review by livio-a — APPROVED** [review]
approve apart from unsolved conflicts

**Comment by livio-a** [comment]
> approve apart from unsolved conflicts

merge main myself and resolved it

**Review by livio-a — APPROVED** [review]

**Pull request merged**